airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalUtils.java

```diff
@@ -162,55 +219,53 @@ public static <STUB, A1, R> ImmutablePair<WorkflowExecution, CompletableFuture<R
 
   /**
    * Loops and waits for the Temporal service to become available and returns a client.
-   *
+   * <p>
    * This function uses a supplier as input since the creation of a WorkflowServiceStubs can result in
    * connection exceptions as well.
    */
   public static WorkflowServiceStubs getTemporalClientWhenConnected(
                                                                     final Duration waitInterval,
                                                                     final Duration maxTimeToConnect,
                                                                     final Duration waitAfterConnection,
-                                                                    final Supplier<WorkflowServiceStubs> temporalServiceSupplier) {
-    LOGGER.info("Waiting for temporal server...");
+                                                                    final Supplier<WorkflowServiceStubs> temporalServiceSupplier,
+                                                                    final String namespace) {
+    log.info("Waiting for temporal server...");
 
-    boolean temporalStatus = false;
+    boolean temporalNamespaceInitialized = false;
     WorkflowServiceStubs temporalService = null;
     long millisWaited = 0;
 
-    while (!temporalStatus) {
+    while (!temporalNamespaceInitialized) {
       if (millisWaited >= maxTimeToConnect.toMillis()) {
         throw new RuntimeException("Could not create Temporal client within max timeout!");
       }
 
-      LOGGER.warn("Waiting for default namespace to be initialized in temporal...");
+      log.warn("Waiting for namespace {} to be initialized in temporal...", namespace);
       Exceptions.toRuntime(() -> Thread.sleep(waitInterval.toMillis()));
       millisWaited = millisWaited + waitInterval.toMillis();
 
       try {
         temporalService = temporalServiceSupplier.get();
-        temporalStatus = getNamespaces(temporalService).contains("default");
+        final var namespaceInfo = getNamespaceInfo(temporalService, namespace);
```

Comment:
We used to call `listNamespaces` to check if our temporal service was online, but this call isn't allowed in Temporal Cloud. I think it's much more direct to call `describeNamespace` and check to see if `isInitialized()` is true, so that's how we check for service initialization now.